device.py

```diff
@@ -1,8 +1,38 @@
 from clip import Clip
 
 
-class Device():
+class DeviceOutput:
+    def __init__(self, method, name=None):
+        self.method = method
+        self.name = name or method.__name__
+        self.__doc__ = method.__doc__
 
+    def get_mapping(self, inst):
+        return inst.mapping
+
+    def __get__(self, inst, cls=None):
+        mapping = self.get_mapping(inst)
+        return mapping.setdefault(self.name, self.method(inst))
+
+    def __set__(self, inst, value):
+        mapping = self.get_mapping(inst)
+        mapping[self.name] = value
+        # inst.update_lookup()
+
+    def __delete__(self, inst):
```

Comment:
**Question**: should deleting even be supported?
- In the old implementation, setters and getters were implemented, but deleters were not
- I cannot think of any reason why one would need to be able to delete values (except for a possible downgrade to an older device-format, which is something that is rather unlikely to be implemented)